Mathlib/MeasureTheory/Measure/FiniteMeasure.lean

```diff
@@ -539,6 +549,17 @@ theorem tendsto_iff_forall_lintegral_tendsto {γ : Type*} {F : Filter γ} {μs :
     ENNReal.toNNReal_coe]
 #align measure_theory.finite_measure.tendsto_iff_forall_lintegral_tendsto MeasureTheory.FiniteMeasure.tendsto_iff_forall_lintegral_tendsto
 
+/-- Weak limits of finite Borel measures are unique. -/
```

Comment:
I am planning to introduce the Lévy-Prokhorov metric in #8719, and with it eventually provide a `MetrizableSpace` instance on finite Borel measures on separable metric spaces (and the present PR is meant to slightly facilitate some lemmas there as well). That generality is by far the most important case in my opinion, but having one formulation of uniqueness of limits without assuming separability and metrizability at least explicitly sounded like a reasonable additional lemma.

Because of this, I was unsure about recording instances here yet --- I don't understand typeclass instance loops well enough. Specifically, is it not a problem if in the most important case there is a direct `T2Space` instance and one derived from `MetrizableSpace`?

It sounds like a good idea anyway to write a proof of `T2Space (FiniteMeasure Ω)`, and if it is safe to record that as an instance, I will be happy to do so.

Reply:
There is now `MeasureTheory.FiniteMeasure.t2Space` for the lemma, but not yet as an instance. I refactored the proof here so that it uses the `t2Space` lemma.

Reply:
There is no issue with instance loops in Mathlib4. But even here, there wouldn't be any loop, just two different ways to derive a T2 instance. So, you can safely make it an instance!